kufar/books/main.go

```diff
@@ -80,9 +88,25 @@ func main() {
 	if err != nil {
 		log.Fatalf("error connecting to db: %v", err)
 	}
-
 	defer db.Close()
 
+	var bk Book
+
+	go func() {
+
+		tick := time.NewTicker(time.Second * 5)
+
+		for {
+			select {
+			case <-tick.C:
+				bk.Time.Lock()
+				bk.Time.Time = append(bk.Time.Time, time.Now())
```

Comment:
Mutex MUST not be copied. In your case, you making a copy here. I suggest you remove your `TimeStruct`. And just add time slice with mutex directly to `Database` struct. And create a method on that struct that will make this work.